File: src/main/java/com/devspace/conexfy/dtos/ConConnectionResponseDTO.java
package com.devspace.conexfy.dtos;

import java.util.Map;

import com.devspace.conexfy.enums.ConAuthTypeEnum;
import com.devspace.conexfy.enums.ConHttpMethodEnum;

public record ConConnectionResponseDTO(
        Long id,
        String name,
        String description,
        ConHttpMethodEnum method,
        ConAuthTypeEnum authType,
        Map<String, String> headersJson,
        String url,
        Map<String, String> pathVarsJson,
        Map<String, String> queryParamsJson,
        String body) {
}

File: src/main/kotlin/com/devspace/conexfy/dtos/ConConnectionResponseDTO.kt
package com.devspace.conexfy.dtos

import com.devspace.conexfy.enums.ConAuthTypeEnum
import com.devspace.conexfy.enums.ConHttpMethodEnum

data class ConConnectionResponseDTO(
    val id: Long?,
    val name: String?,
    val description: String?,
    val method: ConHttpMethodEnum?,
    val authType: ConAuthTypeEnum?,
    val headersJson: MutableMap<String, String>?,
    val url: String?,
    val pathVarsJson: MutableMap<String, String>?,
    val queryParamsJson: MutableMap<String, String>?,
    val body: String
)
